### PR TITLE
Fix choices_as_values deprecation message

### PR DESCRIPTION
**Pull request by jordisala1991**
I am targeting this branch, because this is BC.

## Changelog

```markdown
### Fixed
- Deprecation message on SecurityRolesType about `choices_as_values`
```

## Subject

Seems like the deprecations build is failing because of this.
Link: https://travis-ci.org/sonata-project/SonataUserBundle/jobs/334884037#L798

Probably this makes this bundle unusable with SF 4, that's why I am tagging it with patch

### CODE REVIEW

File: src/Form/Type/SecurityRolesType.php [hunk 1]
```diff
@@ -135,7 +137,7 @@ public function configureOptions(OptionsResolver $resolver): void
         ]);
 
         // Symfony 2.8 BC
-        if ($resolver->isDefined('choices_as_values')) {
+        if (method_exists(FormTypeInterface::class, 'setDefaultOptions')) {
```
Comment (greg0ire) [inline src/Form/Type/SecurityRolesType.php]: Can you explain the fix?

File: tests/Form/Type/SecurityRolesTypeTest.php [hunk 2]
```diff
@@ -94,25 +98,6 @@ public function testSubmitWithHiddenRoleData(): void
         $this->assertContains('ROLE_SUPER_ADMIN', $form->getData());
     }
 
-    public function testChoicesAsValues(): void
```
Comment (greg0ire) [inline tests/Form/Type/SecurityRolesTypeTest.php]: Why is this removed?

File: src/Form/Type/SecurityRolesType.php (hunk 1 above)
Comment (kunicmarko20) [reply src/Form/Type/SecurityRolesType.php]: choices_as_values is removed in Symfony 4

Comment (greg0ire) [reply src/Form/Type/SecurityRolesType.php]: Which means `$resolver->isDefined('choices_as_values')` will return `false`, which means `$resolver->setDefault('choices_as_values', true);` will not be set to `true`, which… makes sense if that option is removed?

Comment (kunicmarko20) [reply src/Form/Type/SecurityRolesType.php]: The default value of `choices_as_values` is false and that shows deprecation notices, but we do not have that option in SF4, so he checks if it is < SF3 and sets it to true to avoid deprecation notices. Am I correct @jordisala1991 ?

Comment (jordisala1991) [reply src/Form/Type/SecurityRolesType.php]: Yes, that's also probably. I just wanted to remove the deprecations on Sf3, but wasn't sure if this would work on Sf4

File: tests/Form/Type/SecurityRolesTypeTest.php (hunk 2 above)
Comment (jordisala1991) [reply tests/Form/Type/SecurityRolesTypeTest.php]: Because this is tested already on the test above, but conditionally only if there are choices_as_values option

File: src/Form/Type/SecurityRolesType.php [hunk 3]
```diff
@@ -135,7 +137,7 @@ public function configureOptions(OptionsResolver $resolver): void
...
             $resolver->setDefault('choices_as_values', true);
```
Comment (greg0ire) [inline src/Form/Type/SecurityRolesType.php]: The coverage decreased and coveralls says this is no longer covered by tests… is that true and normal?

Comment (kunicmarko20) [reply src/Form/Type/SecurityRolesType.php]: It is covered on 2.8 https://coveralls.io/jobs/33261397/source_files/324132516#L140, because `setDefaultOptions` exist only there.

**Comment by greg0ire** [comment]
Thanks @jordisala1991 !
